packages/cms-lib/fileManager.js

```diff
@@ -170,68 +174,118 @@ async function getAllPagedFiles(portalId, folderPath) {
  * @param {object} options
  */
 async function fetchFolderContents(portalId, folder, dest, options) {
-  const files = await getAllPagedFiles(portalId, folder.full_path);
+  const files = await getAllPagedFiles(portalId, folder.id, options);
   for (const file of files) {
     await downloadFile(portalId, file, dest, options);
   }
 
-  const { objects: folders } = await getFoldersByPath(
-    portalId,
-    folder.full_path
-  );
+  const { objects: folders } = await getFolders(portalId, folder.id);
   for (const folder of folders) {
     const nestedFolder = path.join(dest, folder.name);
     await fetchFolderContents(portalId, folder, nestedFolder, options);
   }
 }
 
 /**
- * Fetch a file/folder and write to local file system.
+ * Download a folder and write to local file system.
  *
  * @param {number} portalId
  * @param {string} src
  * @param {string} dest
+ * @param {object} file
  * @param {object} options
  */
-async function downloadFileOrFolder(portalId, src, dest, options) {
-  const { file, folder } = await getStat(portalId, src);
-  if (file) {
+async function downloadFolder(portalId, src, dest, folder, options) {
+  try {
+    const rootPath =
+      dest === getCwd()
+        ? convertToLocalFileSystemPath(path.resolve(dest, folder.name))
+        : dest;
     logger.log(
-      'Fetching file from "%s" to "%s" in the File Manager of portal %s',
+      'Fetching folder from "%s" to "%s" in the File Manager of portal %s',
       src,
-      dest,
+      rootPath,
       portalId
     );
-    try {
-      await downloadFile(portalId, file, dest, options);
-      logger.success(
-        'Completed fetch of file "%s" to "%s" from the File Manager',
-        src,
-        dest
-      );
-    } catch (err) {
-      logErrorInstance(err);
-    }
-  } else if (folder) {
+    await fetchFolderContents(portalId, folder, rootPath, options);
+    logger.success(
+      'Completed fetch of folder "%s" to "%s" from the File Manager',
+      src,
+      dest
+    );
+  } catch (err) {
+    logErrorInstance(err);
+  }
+}
+
+/**
+ * Download a single file and write to local file system.
+ *
+ * @param {number} portalId
+ * @param {string} src
+ * @param {string} dest
+ * @param {object} file
+ * @param {object} options
+ */
+async function downloadSingleFile(portalId, src, dest, file, options) {
+  if (file.archived) {
```

Comment:
Should there be a check against `options` here to determine if the flag is passed?